src/main/java/htsjdk/variant/vcf/VCFPathReader.java

```diff
@@ -0,0 +1,183 @@
+
+    private final FeatureReader<VariantContext> reader;
+
+    /**
+     * Returns true if the given path appears to be a BCF file.
+     */
+    public static boolean isBCF(final Path path) {
+        return path.toUri().getRawPath().endsWith(".bcf");
+    }
+
+    /**
+     * Returns the SAMSequenceDictionary from the provided VCF file.
+     */
+    public static SAMSequenceDictionary getSequenceDictionary(final Path path) {
+        final SAMSequenceDictionary dict = new VCFPathReader(path, false).getFileHeader().getSequenceDictionary();
+        CloserUtil.close(path);
+        return dict;
+    }
+
+    /**
+     * Constructs a VCFFileReader that requires the index to be present.
+     */
+    public VCFPathReader(final Path path) {
+        this(path, true);
+    }
+
+    /**
+     * Constructs a VCFFileReader with a specified index.
+     */
+    public VCFPathReader(final Path path, final Path indexPath) {
+        this(path, indexPath, true);
+    }
+
+    /**
+     * Allows construction of a VCFFileReader that will or will not assert the presence of an index as desired.
+     */
+    public VCFPathReader(final Path path, final boolean requireIndex) {
+        // Note how we deal with type safety here, just casting to (FeatureCodec)
+        // in the call to getFeatureReader is not enough for Java 8.
+        FeatureCodec<VariantContext, ?> codec = isBCF(path) ? new BCF2Codec() : new VCFCodec();
+        this.reader = AbstractFeatureReader.getFeatureReader(
+                path.toUri().toString(),
+                codec,
+                requireIndex);
+    }
+
+    /**
+     * Allows construction of a VCFFileReader with a specified index path.
+     */
+    public VCFPathReader(final Path path, final Path indexPath, final boolean requireIndex) {
+        // Note how we deal with type safety here, just casting to (FeatureCodec)
+        // in the call to getFeatureReader is not enough for Java 8.
+        FeatureCodec<VariantContext, ?> codec = isBCF(path) ? new BCF2Codec() : new VCFCodec();
+        this.reader = AbstractFeatureReader.getFeatureReader(
+                path.toUri().toString(),
+                indexPath.toUri().toString(),
+                codec,
+                requireIndex);
+    }
+
+
+    /**
+     * Parse a VCF file and convert to an IntervalList The name field of the IntervalList is taken from the ID field of the variant, if it exists. if not,
+     * creates a name of the format interval-n where n is a running number that increments only on un-named intervals
+     *
+     * @param path
+     * @return
+     */
+    public static IntervalList fromVcf(final Path path) {
+        return fromVcf(path, false);
+    }
+
+    public static IntervalList fromVcf(final Path path, final boolean includeFiltered) {
+        final VCFPathReader vcfReader = new VCFPathReader(path, false);
+        final IntervalList intervalList = fromVcf(vcfReader, includeFiltered);
+        vcfReader.close();
```

Comment:
Using try-with-resources would eliminate the need to call `close()` here

Reply:
done.